cri-purge.sh

```diff
@@ -115,45 +131,52 @@ __process_images() {
   echo "Total Images: ${TOTAL_CRI_IMAGES} Unique Images Names: ${TOTAL_UNIQUE_IMAGE_NAMES}"
   echo
   COUNT=0
-  for IMAGE_NAME in ${UNIQUE_CRI_IMAGE_NAMES}; # Do not quote UNIQUE_CRI_IMAGE_NAMES, breaks for loop
+  while read IMAGE_NAME; # Do not quote UNIQUE_CRI_IMAGE_NAMES, breaks for loop
   do
     ((COUNT=COUNT+1))
     echo -n "${COUNT} / ${TOTAL_UNIQUE_IMAGE_NAMES} : Image: ${IMAGE_NAME}"
 
     # Find all versions of this IMAGE_NAME
-    mapfile -t IMAGES <<< "$(echo "${CRI_IMAGES}" | grep "${IMAGE_NAME}")"
+    mapfile -t IMAGES <<< "$(grep "${IMAGE_NAME}" /tmp/CRI_IMAGES)"
     NUM_IMAGES=${#IMAGES[@]}
 
     # If only 1 version detected, keep it.
     if [[ ${NUM_IMAGES} -eq 1 ]]; then
-      echo " - Keep TAG: $(echo "${IMAGES[0]}" | awk '{ printf "%s (%s)\n", $2, $4 }')"
+      TAG_REF=$(__get_tag_ref 0)
+      echo " - Keep TAG: $(echo "${IMAGES[0]}" | awk "{ printf \"%s (%s)\n\", $TAG_REF, \$4 }")"
     else
       # print last line of the array, should be one to keep.
-      echo " - Keep TAG: $(printf %s\\n "${IMAGES[@]: -1}"| awk '{ printf "%s (%s)\n", $2, $4 }')"
+      TAG_REF=$(__get_tag_ref 1)
+      echo " - Keep TAG: $(printf %s\\n "${IMAGES[@]: -1}"| awk "{ printf \"%s (%s)\n\", $TAG_REF, \$4 }")"
 
       for (( i=0; i<$(( ${#IMAGES[@]}-1 )); i++ )) 
       do
         # Remove image if $1 == "PURGE"
+        TAG_REF=$(__get_tag_ref $i)
         if [ "${1^^}" == "PURGE" ]; then
-          echo "- Purge TAG: $( echo "${IMAGES[$i]}" | awk '{ printf "%s (%s)\n", $2, $4 }')"
-
-          # Remove the Specific Image:TAG
-          ${CRI_CMD} rmi "$(echo "${IMAGES[$i]}" |awk '{ printf "%s:%s\n", $1, $2 }')" > /dev/null 2>&1
+          echo "- Purge TAG: $( echo "${IMAGES[$i]}" | awk "{ printf \"%s (%s)\n\", $TAG_REF, \$4 }")"
+
+          # Remove the Specific Image:TAG, or by a hash (if tag=<none>)
+          if [ "$TAG_REF" = "\$3" ]; then
+            ${CRI_CMD} rmi "$(echo "${IMAGES[$i]}" |awk '{ printf "%s\n", $3 }')" > /dev/null 2>&1
+          else
+            ${CRI_CMD} rmi "$(echo "${IMAGES[$i]}" |awk '{ printf "%s:%s\n", $1, $2 }')" > /dev/null 2>&1
+          fi
         else
-          echo "- Purgeable TAG: $( echo "${IMAGES[$i]}" | awk '{ printf "%s (%s)\n", $2, $4 }')"
+          echo "- Purgeable TAG: $( echo "${IMAGES[$i]}" | awk "{ printf \"%s (%s)\n\", $TAG_REF, \$4 }")"
         fi
       done
       echo
     fi
-  done
+  done < /tmp/UNIQUE_CRI_IMAGE_NAMES
 }
 
 ###[ Main Section ]##########################################################
 
 # Confirm crictl is installed
-if ! command -v ${CRI_CMD} >/dev/null 2>&1; then
+if ! command -v ${CRI_CMD} >/dev/null 2>&1 || ! command -v ${CRIINFO_CMD} >/dev/null 2>&1 ; then
```

Comment:
I had to change that to "&&" instead of "||" as it would fail if either one was missing. I think it would fail only if both are missing.